lib/iris/_data_manager.py

```diff
@@ -382,6 +411,26 @@ def dtype(self):
         return result
 
     @property
+    def fill_value(self):
+        return self._fill_value
+
+    @fill_value.setter
+    def fill_value(self, fill_value):
+        if fill_value is not None:
+            # Convert the given value to the dtype of the data manager.
+            fill_value = np.asarray([fill_value])[0]
+            target_dtype = self.dtype
+            if fill_value.dtype.kind == 'f' and target_dtype.kind in 'biu':
+                # Perform rounding when converting floats to ints.
+                fill_value = np.rint(fill_value)
```

Comment:
Yeah, it's probably better if this PR doesn't introduce any functionality changes anyway.